engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2324,17 +2357,11 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
       int i = 0;
       try {
         while (++i < args.length) {
-          app = getApplication(args[i]);
+          app = findApplicationReportFromAppNameOrId(args[i]);
           if (app == null) {
-
-            /*
-             * try once again with application name type.
-             */
-            app = getApplicationByName(args[i]);
-            if (app == null) {
-              throw new CliException("Streaming application with id or name " + args[i] + " is not found.");
-            }
+            throw new CliException("Streaming application with id or name " + args[i] + " is not found.");
```

Comment:
Why do we throw exception and stop processing the kill command as opposed to what we decided to do for the shutdown command e.g. continue processing the remaining arguments? In that case we can also achieve some code reuse between shutdown and kill commands - building the app report list etc can be common code.